src/main/webapp/js/index/index.js

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* ==========================================================================
+   ROULETTE
+   ========================================================================== */
+function query(queryStr) {
+    const errorEl = document.getElementById("error");
+    fetch(`/query?${queryStr}`, { method: "POST" })
+        .then((response) => response.json())
+        .then((response) => {
+            console.log(response);
+            if (response.status === "OK")
+                redirectToUrl('results.html');
+            else if (response.status === "INVALID_REQUEST") throw "Invalid request";
+            else if (response.status === "ZERO_RESULTS") throw "No results";
+            else if (response.status === "NO_REROLLS") throw "No re-rolls left";
```

Comment:
what's the difference between index.js and query.js? This code appears twice. Can we unify these?

Reply:
query.js is outdated and split up into index.js and results.js, I don't think it's part of this branch though

Reply:
can we fix the code here so that there's only one appearance of this function implementation? I think it's better if we avoid duplicating the function in the new code here.